tools/test-proxy/transition-scripts/generate-assets-json.ps1

```diff
@@ -89,28 +121,183 @@ Function Get-Repo-Language {
   return $lang
 }
 
-Test-Exe-In-Path -ExeToLookFor "git"
-$AssetsRepoPrefixPath = Get-Repo-Language
+Function Get-Repo-Root {
+  [string] $currentDir = Get-Location
+  # -1 to strip off the trialing directory separator
+  return $currentDir.Substring(0, $currentDir.LastIndexOf("sdk") - 1)
+}
+
+Function New-Assets-Json-File {
+  param(
+    [Parameter(Mandatory = $true)]
+    [string] $Language
+  )
+  $AssetsRepoPrefixPath = $Language
+
+  [string] $currentDir = Get-Location
+
+  $sdkDir = "$([IO.Path]::DirectorySeparatorChar)sdk$([IO.Path]::DirectorySeparatorChar)"
+
+  # if we're not in a <reporoot>/sdk/<ServiceDirectory> or deeper then this script isn't
+  # being run in the right place
+  if (-not $currentDir.contains($sdkDir)) {
+    Write-Error "This script needs to be run at an sdk/<ServiceDirectory> or deeper."
+    exit 1
+  }
+
+  $TagPrefix = $currentDir.Substring($currentDir.LastIndexOf("sdk") + 4)
+  $TagPrefix = $TagPrefix.Replace("\", "/")
+  $TagPrefix = "$($AssetsRepoPrefixPath)/$($TagPrefix)"
+  [Assets]$Assets = [Assets]::new($AssetsRepoPrefixPath, $TagPrefix)
 
-[string] $currentDir = Get-Location
+  $AssetsJson = $Assets | ConvertTo-Json
 
-$sdkDir = "$([IO.Path]::DirectorySeparatorChar)sdk$([IO.Path]::DirectorySeparatorChar)"
+  $AssetsFileName = Join-Path -Path $currentDir -ChildPath "assets.json"
+  Write-Host "Writing file $AssetsFileName with the following contents"
+  Write-Host $AssetsJson
+  $Assets | ConvertTo-Json | Out-File $AssetsFileName
 
-# if we're not in a <reporoot>/sdk/<ServiceDirectory> or deeper then this script isn't
-# being run in the right place
-if (-not $currentDir.contains($sdkDir)) {
-  Write-Error "This script needs to be run at an sdk/<ServiceDirectory> or deeper."
-  exit 1
+  return $AssetsFileName
 }
 
-$TagPrefix = $currentDir.Substring($currentDir.LastIndexOf("sdk") + 4)
-$TagPrefix = $TagPrefix.Replace("\", "/")
-$TagPrefix = "$($AssetsRepoPrefixPath)/$($TagPrefix)"
-[Assets]$Assets = [Assets]::new($AssetsRepoPrefixPath, $TagPrefix)
+# Invoke the proxy command and echo the output.
+Function Invoke-ProxyCommand {
+  param(
+    [string] $TestProxyExe,
+    [string] $CommandArgs
```

Comment:
So this is just a transition script, so the usage will be limited. I don't know that the error will pop up, but we we should pass command args as an array of strings. 
```suggestion
    [string[]] $CommandArgs
```
And then just pass the array directly to the `&` .

I give this advice because what happens if we were to have a space in the command itself? Like if a target folder had a space in it?

Reply:
cwd will evaluate to  `"C:/repo/sdk-for-python/sdk/hello world"`. Split, and we'll pass a command that looks like:

`test-proxy ""C:/repo/sdk-for-python/sdk/hello" "world""`

Will this _ever_ happen? I don't think so, but it's also a super minor fix.